src/app/component/auth/components/sign-up/sign-up.component.spec.ts

```diff
@@ -34,22 +42,265 @@ describe('SignUpComponent', () => {
       ],
 
       providers: [
-        AuthService,
+        { provide: AuthService, useValue: authServiceMock },
         { provide: MatDialogRef, useClass: MatDialogRefMock },
         { provide: AuthServiceConfig, useFactory: provideConfig },
+        { provide: Router, useValue: routerSpy },
+        UserOwnSignUpService,
       ],
       schemas: [CUSTOM_ELEMENTS_SCHEMA, NO_ERRORS_SCHEMA]
     })
-    .compileComponents();
+      .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(SignUpComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
+    socialUserMock = new SocialUser();
+    socialUserMock.email = '';
+    socialUserMock.firstName = '';
+    socialUserMock.authorizationCode = '';
+    socialUserMock.id = '';
+    socialUserMock.name = '';
+    socialUserMock.photoUrl = '';
+    socialUserMock.authToken = '';
+    authServiceMock = jasmine.createSpyObj('AuthService', ['signIn']);
+    authServiceMock.signIn = (providerId: string, opt?: LoginOpt) => new Promise<SocialUser>((resolve) => {
+      resolve(socialUserMock);
+    });
+  });
+
+  describe('Basic tests', () => {
+    beforeEach(() => {
+      // @ts-ignore
+      spyOn(component.pageName, 'emit');
+    });
+
+    it('should create SignUpComponent', () => {
+      expect(component).toBeTruthy();
+    });
+
+    it('should call closeSignUpWindow ', () => {
+      // @ts-ignore
+      const spy = spyOn(component.matDialogRef, 'close').and.callThrough();
+      // @ts-ignore
+      component.closeSignUpWindow();
+      expect(spy).toHaveBeenCalled();
+    });
+
+    it('should call openSignUpPopup', () => {
+      // @ts-ignore
```

Comment:
Why you need this ignore?

Reply:
> Why you need this ignore?

we use it to test private methods. are there other constructions for this?